beam-core/Database/Beam/Backend/Types.hs

```diff
@@ -25,34 +30,54 @@ import GHC.Generics
 import GHC.TypeLits
 import GHC.Types
 
+data BeamRowError
```

Comment:
use a newtype or get rid of this error. Does the Exception instance imply this type will be thrown?